project_types/base/project.py

```diff
@@ -420,6 +421,17 @@ def update_project_on_firebase(self, project_ref: FbReference, fb_project: fireb
         assert self.project.tutorial_id is not None, "Tutorial is required before project can be pushed to firebase"
         assert self.project.tutorial is not None, "Tutorial is required before project can be pushed to firebase"
 
+        unique_contributors_count = (
+            MappingSession.objects.filter(
+                project_task_group__project_id=self.project.pk,
```

Comment:
```suggestion
                project_task_group__in=ProjectTaskGroup.objects.filter(project=self.project)
```
Let's use this to avoid large join.